tests/integration/models/application-membership.spec.ts

```diff
@@ -0,0 +1,371 @@
+			before(async function () {
+				membership = await balena.models.application.membership.create({
+					application: this.application.id,
+					username: credentials.member.username,
+					roleName: 'developer',
+				});
+			});
+			after(async function () {
+				await balena.models.application.membership.remove(membership!.id);
+			});
+
+			describe('balena.models.application.membership.changeRole()', function () {
+				it(`should not be able to change an application membership to an unknown role`, async function () {
+					const promise = balena.models.application.membership.changeRole(
+						membership!.id,
+						'unknown role',
+					);
+					await expect(promise).to.be.rejected.and.eventually.have.property(
+						'code',
+						'BalenaApplicationMembershipRoleNotFound',
+					);
+				});
+
+				it(`should not be able to change an application membership to an operator role on a starter application `, async function () {
+					const promise = balena.models.application.membership.changeRole(
+						membership!.id,
+						'operator',
+					);
+
+					await expect(promise).to.be.rejectedWith(
+						'It is necessary that each user application membership that has an application membership role that has a name (Auth) that is not equal to "developer" and is not equal to "observer", has an application that has an application type that has a slug that is not equal to "microservices-starter"',
+					);
+				});
+
+				const roleChangeTest = (
+					rolenName: BalenaSdk.ApplicationMembershipRoles,
+				) =>
+					it(`should be able to change an application membership to "${rolenName}"`, async function () {
+						await balena.models.application.membership.changeRole(
+							membership!.id,
+							rolenName,
+						);
+
+						membership = await balena.models.application.membership.get(
+							membership!.id,
+							{
+								$select: 'id',
+								$expand: {
+									user: {
+										$select: 'username',
+									},
+									application_membership_role: {
+										$select: 'name',
+									},
+								},
+							},
+						);
+						expect(membership).to.have.nested.property(
+							'user[0].username',
+							credentials.member.username,
+						);
+						expect(membership).to.have.nested.property(
+							'application_membership_role[0].name',
+							rolenName,
+						);
+					});
+
+				roleChangeTest('observer');
+				roleChangeTest('developer');
+			});
+
+			describe('balena.models.application.membership.remove()', function () {
+				it(`should be able to remove a developer`, async function () {
+					await balena.models.application.membership.remove(membership!.id);
+
+					const promise = balena.models.application.membership.get(
+						membership!.id,
+					);
+
+					await expect(promise).to.be.rejectedWith(
+						'Application Membership not found',
+					);
+				});
+			});
+		});
+	});
+
+	describe('balena.models.application.membership.getAll()', function () {
```

Comment:
I guess that this could also be `parallel` but not a blocker. Eg:
```
decribe('given a membership',
  before(...)
  after(...)
  parallel('balena.models.application.membership.getAll()', function () {
    let ctx
    before(function(){ ctx = this; });
```


I guess you can just move this to the `describe('given a membership [read operations]'` right below so that we also re-use the same membership for the tests.

Reply:
actually much better! thank you !